front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -161,57 +102,49 @@ export function AgentMessage({
   owner,
 }: AgentMessageProps) {
   const { isDark } = useTheme();
-  const [streamedAgentMessage, setStreamedAgentMessage] =
-    useState<AgentMessageType>(message);
 
-  const [isRetryHandlerProcessing, setIsRetryHandlerProcessing] =
-    useState<boolean>(false);
-
-  const [references, setReferences] = useState<{
-    [key: string]: MarkdownCitation;
-  }>({});
-
-  const [activeReferences, setActiveReferences] = useState<
-    { index: number; document: MarkdownCitation }[]
-  >([]);
-  const [isCopied, copy] = useCopyToClipboard();
-
-  const isGlobalAgent = useMemo(() => {
-    return Object.values(GLOBAL_AGENTS_SID).includes(
-      message.configuration.sId as GLOBAL_AGENTS_SID
-    );
-  }, [message.configuration.sId]);
+  const [state, dispatch] = React.useReducer(messageReducer, {
```

Comment:
nitpick but this can be `useReducer(messageReducer, message, getInitialState)`  to avoid re-creating object on ever render?